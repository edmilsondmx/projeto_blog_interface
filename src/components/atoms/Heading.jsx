import styled from "styled-components";

const Heading = styled.div`
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    color: ${(props) => props.theme.colors.text};
    position: relative;
    margin-bottom: 25px;
    padding-bottom: 8px;
    border-bottom: 1px solid ${(props) => props.theme.colors.border};

    &::after {
      content: "";
      position: absolute;
      left: 0;
      bottom: -3px;
      background-color: ${(props) => props.theme.colors.main};
      height: 5px;
      width: 70px;
    }
  }
  h1 {
    padding-bottom: 25px;
  }
  h2 {
    padding-bottom: 16px;
  }
`;

export default Heading;
