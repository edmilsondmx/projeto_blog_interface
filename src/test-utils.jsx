import React from "react";
import { BrowserRouter } from "react-router-dom";
import ThemeProvider from "./styles/ThemeProvider";
import { render } from "@testing-library/react";

const TestProvider = ({ children }) => (
  <BrowserRouter basename="/">
    <ThemeProvider>{children}</ThemeProvider>
  </BrowserRouter>
);

export * from "@testing-library/react";

const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

export { customRender as render };
