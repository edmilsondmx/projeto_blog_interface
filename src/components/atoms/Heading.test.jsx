import React from "react";
import { render } from "../../test-utils";
import Heading from "./Heading";

it("match snapshot", () => {
  const { asFragment } = render(<Heading>Title</Heading>);

  expect(asFragment()).toMatchSnapshot();
});
