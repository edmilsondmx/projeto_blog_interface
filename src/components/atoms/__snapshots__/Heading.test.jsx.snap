// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB ilIZta"
  >
    Title
  </div>
</DocumentFragment>
`;
